### PR TITLE
Validate Content status and assigned user

### PR DESCRIPTION
**Pull request by elisabeth-shakespeare**

### CODE REVIEW

File: src/Aquifer.API/Endpoints/Resources/Content/MachineTranslation/Update/Endpoint.cs [hunk 1]
```diff
@@ -19,9 +20,12 @@ public override async Task HandleAsync(Request req, CancellationToken ct)
         var user = await userService.GetUserFromJwtAsync(ct);
         var existingMt = await dbContext.ResourceContentVersionMachineTranslations
             .AsTracking()
+            .Include(rcvmt => rcvmt.ResourceContentVersion)
+            .ThenInclude(rcv => rcv.ResourceContent)
             .FirstOrDefaultAsync(x => x.Id == req.Id, ct);
 
-        if (existingMt is null || existingMt.UserId != user.Id)
+        if (existingMt is null || existingMt.ResourceContentVersion.AssignedUserId != user.Id ||
```
Comment (elisabeth-shakespeare) [inline src/Aquifer.API/Endpoints/Resources/Content/MachineTranslation/Update/Endpoint.cs:27]: Should I clean some of this up while I'm at it? maybe do a ValidateXXX method, and an UpdateXXX method for the other changes?

Comment (jwinston-bn) [reply src/Aquifer.API/Endpoints/Resources/Content/MachineTranslation/Update/Endpoint.cs:27]: I think we can do these checks in the `FirstOrDefaultAsync` and then just check `existingMt is null`.

File: src/Aquifer.API/Endpoints/Resources/Content/MachineTranslation/Update/Endpoint.cs [hunk 2]
```diff
@@ -19,9 +20,14 @@ public override async Task HandleAsync(Request req, CancellationToken ct)
         var user = await userService.GetUserFromJwtAsync(ct);
         var existingMt = await dbContext.ResourceContentVersionMachineTranslations
             .AsTracking()
-            .FirstOrDefaultAsync(x => x.Id == req.Id, ct);
-
-        if (existingMt is null || existingMt.UserId != user.Id)
+            .Include(rcvmt => rcvmt.ResourceContentVersion)
+            .ThenInclude(rcv => rcv.ResourceContent)
```
Comment (jwinston-bn) [inline src/Aquifer.API/Endpoints/Resources/Content/MachineTranslation/Update/Endpoint.cs:24]: Do we need these?

File: src/Aquifer.API/Endpoints/Resources/Content/MachineTranslation/Update/Endpoint.cs [hunk 3]
```diff
@@ -19,9 +20,14 @@ public override async Task HandleAsync(Request req, CancellationToken ct)
...
+            .FirstOrDefaultAsync(
+                x => x.Id == req.Id && (x.ResourceContentVersion.AssignedUserId == user.Id ||
+                                        x.ResourceContentVersion.ResourceContent.Status == ResourceContentStatus.TranslationEditorReview),
```
Comment (jwinston-bn) [inline src/Aquifer.API/Endpoints/Resources/Content/MachineTranslation/Update/Endpoint.cs:27]: This should be `&&` not `||` I think.

File: src/Aquifer.API/Endpoints/Resources/Content/MachineTranslation/Update/Endpoint.cs (hunk 2 above)
Comment (elisabeth-shakespeare) [reply src/Aquifer.API/Endpoints/Resources/Content/MachineTranslation/Update/Endpoint.cs:24]: I had thought we did so we can access the props of them, but I'll double check and make sure

Comment (elisabeth-shakespeare) [reply src/Aquifer.API/Endpoints/Resources/Content/MachineTranslation/Update/Endpoint.cs]: oh cool. I don't need them.

**Review by jwinston-bn — APPROVED** [review]

**Pull request merged**